R/main.R

```diff
@@ -1,118 +1,149 @@
-
-        scenObjects[[i]] <- currScenInfo
-        i <- i + 1
-
-        # Loop over all TAU options and run Lagged
-        for(share in levels.LAGSHARE) {
-          scenName <- getScenName(SCENARIO, "Lagged", share, agFor, agForNonPast, crop)
-
-          currScenInfo <- ScenarioInfo(aScenario = SCENARIO,
-                                       aExpectationType = "Lagged",
-                                       aLinearYears = NULL,
-                                       aLaggedShareOld = share,
-                                       aLogitUseDefault = FALSE,
-                                       aLogitAgroForest = agFor,
-                                       aLogitAgroForest_NonPasture = agForNonPast,
-                                       aLogitCropland = crop,
-                                       aScenarioName = scenName,
-                                       aFileName = i,
-                                       aOutputDir = aOutputDir)
-
-          scenObjects[[i]] <- currScenInfo
-          i <- i + 1
-        }
-
-        # Loop over all LINYEARS options and run Linear
-        for(linyears in levels.LINYEARS) {
-          scenName <- getScenName(SCENARIO, "Linear", linyears, agFor, agForNonPast, crop)
-          currScenInfo <- ScenarioInfo(aScenario = SCENARIO,
-                                       aExpectationType = "Linear",
-                                       aLinearYears = linyears,
-                                       aLaggedShareOld = NULL,
-                                       aLogitUseDefault = FALSE,
-                                       aLogitAgroForest = agFor,
-                                       aLogitAgroForest_NonPasture = agForNonPast,
-                                       aLogitCropland = crop,
-                                       aScenarioName = scenName,
-                                       aFileName = i,
-                                       aOutputDir = aOutputDir)
-
-          scenObjects[[i]] <- currScenInfo
-          i <- i + 1
-        }
-      }
-    }
-
-  }
+  scenObjects <- Map(gen_ensemble_member,
+                     levels.AGROFOREST, levels.AGROFOREST_NONPASTURE, levels.CROPLAND,
+                     levels.LAGSHARE, levels.LINYEARS,
+                     seq_along(levels.AGROFOREST), aOutputDir) %>%
+    unlist(recursive=FALSE)
 
   # Loop over all scenario configurations and run the model
   foreach(obj = scenObjects) %dopar% {
     message("Starting simulation: ", obj$mFileName)
     run_model(obj)
   }
+  invisible(scenObjects)
+}
+
+#' Generate the ensemble members for a single set of parameters
+#'
+#' This generates one each of the Perfect, Lagged, and Linear scenario types
+#' using the input parameters.  The return value is a list of the three
+#' ScenarioInfo objects for the scenarios generated.
+#'
+#' @param agFor The logit exponent the ag/forest(?) nest
+#' @param agForNonPast The logit exponent for the non-pasture(?) nest
```

Comment:
Yes, this is for the AgForest_NonPasture nest (essentially, the competition between crop, grass/shrub, and forest